Mathlib/Data/List/BigOperators/Basic.lean

```diff
@@ -23,22 +23,45 @@ variable {ι α M N P M₀ G R : Type*}
 
 namespace List
 
-section Monoid
+section Mul
 
-variable [Monoid M] [Monoid N] [Monoid P] {l l₁ l₂ : List M} {a : M}
+variable [Mul M] [One M]
 
 @[to_additive (attr := simp)]
 theorem prod_nil : ([] : List M).prod = 1 :=
   rfl
 #align list.prod_nil List.prod_nil
 #align list.sum_nil List.sum_nil
```

Comment:
I think there is not much point here, keep it assuming `MulOneClass`

Reply:
Ok I have switched it to `MulOneClass` (initially it was `Monoid`)